server/app/services/program/EligibilityNotValidForProgramTypeException.java

```diff
@@ -0,0 +1,12 @@
+package services.program;
+
+/**
+ * EligibilityNotValidForProgramType is thrown when eligibility predicates are attempted to be added
+ * to a Program that has a ProgramType that cannot have eligibility conditions.
+ */
+public class EligibilityNotValidForProgramTypeException extends Exception {
+
+  public EligibilityNotValidForProgramTypeException() {
+    super("Eligibility conditions cannot be set for this ProgramType.");
+  }
```

Comment:
**nitpick (non-blocking, if-minor):** include the program type in the error message (on the theory that it could be something other than common intake in the future)